hub/core/chunk/chunk_compressed_chunk.py

```diff
@@ -1,29 +1,148 @@
 import numpy as np
-from typing import List
+from typing import List, Union, Sequence
 from hub.core.compression import (
     compress_bytes,
     compress_multiple,
     decompress_bytes,
     decompress_multiple,
 )
+from hub.core.sample import Sample  # type: ignore
+from hub.core.fast_forwarding import ffw_chunk
 from hub.core.serialize import bytes_to_text, check_sample_shape
 from hub.core.tiling.sample_tiles import SampleTiles
 from hub.util.casting import intelligent_cast
 from hub.util.compression import get_compression_ratio
 from hub.util.exceptions import SampleDecompressionError
 from .base_chunk import BaseChunk, InputSample
+from hub.core.serialize import infer_chunk_num_bytes
 
 
 class ChunkCompressedChunk(BaseChunk):
+    def __init__(self, *args, **kwargs):
+        super(ChunkCompressedChunk, self).__init__(*args, **kwargs)
+        if self.is_byte_compression:
+            self.decompressed_bytes = bytearray(
+                decompress_bytes(self._data_bytes, self.compression)
+            )
+        else:
+            shapes = [
+                self.shapes_encoder[i] for i in range(self.shapes_encoder.num_samples)
+            ]
+            self.decompressed_samples = decompress_multiple(self._data_bytes, shapes)
+        self._changed = False
+        self._compression_ratio = 2
+
     def extend_if_has_space(self, incoming_samples: List[InputSample]) -> float:
+
         self.prepare_for_write()
         if self.is_byte_compression:
             return self.extend_if_has_space_byte_compression(incoming_samples)
-        return self.extend_if_has_space_img_compression(incoming_samples)
+        return self.extend_if_has_space_image_compression(incoming_samples)
+
+    def extend_if_has_space_byte_compression(
+        self, incoming_samples: Union[Sequence[InputSample], np.ndarray]
+    ):
+        num_samples = 0
+        for i, incoming_sample in enumerate(incoming_samples):
+            serialized_sample, shape = self.serialize_sample(
+                incoming_sample,
+                chunk_compression=self.compression,
+                store_uncompressed_tiles=True,
+            )
+
+            self.num_dims = self.num_dims or len(shape)
+            check_sample_shape(shape, self.num_dims)
+
+            if isinstance(serialized_sample, SampleTiles):
+                incoming_samples[i] = serialized_sample  # type: ignore
+                if self.is_empty:
+                    self.write_tile(serialized_sample)
+                    num_samples += 0.5  # type: ignore
+                    tile = serialized_sample.yield_uncompressed_tile()
+                    self.decompressed_bytes = tile.tobytes()
+                    self._changed = True
+                break
+
+            sample_nbytes = len(serialized_sample)
+
+            recompressed = False  # This flag helps avoid double concatenation
+            if (
+                len(self.decompressed_bytes) + sample_nbytes  # type: ignore
```

Comment:
both image and byte compression are using fairly similar logic for this , can you refactor this out?